edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.CloudProxy/ServiceProxy.cs

```diff
@@ -120,6 +123,21 @@ public async Task<Option<ServiceIdentity>> GetServiceIdentity(string deviceId, s
             return serviceIdentityResult;
         }
 
+        Exception MapException(DeviceScopeApiException ex)
+        {
+            switch (ex.StatusCode)
+            {
+                case HttpStatusCode.Unauthorized:
+                case HttpStatusCode.Forbidden:
+                    return new DeviceInvalidStateException($"Deviced not in scope: [{ex.StatusCode}: {ex.Message}].");
+                case HttpStatusCode.BadRequest:
+                case HttpStatusCode.NotFound:
+                    return new DeviceInvalidStateException($"Deviced not found: [{ex.StatusCode}: {ex.Message}].");
+                default:
+                    return new TimeoutException($"Request failed: [{ex.StatusCode}: {ex.Message}].");
```

Comment:
Why not include the original exception as inner exception? With just ex.Message, we will lose the callstack. Otherwise pls make the exception text more verbose, so that we know exactly where it was thrown.